triggers/service/authentication.md

```diff
@@ -0,0 +1,51 @@
+# Trigger Service Authentication
+
+The trigger service is a client of the
+[reference authentication service](https://github.com/digital-asset/ref-ledger-authenticator),
+which we'll just call the "auth service".
+It is used to obtain tokens used to access the ledger on behalf of trigger service users.
+
+## Auth service concepts
+
+The auth service has the following important entities:
+ * user
+ * service account (SA)
+ * SA credential
+ * SA token
+
+The SA token is a JWT token eventually used for ledger access.
+
+In general, each entity in the list can have a one-to-many relationship with an entity later in the list.
+For our purposes, users and service accounts correspond one-to-one and represent parties using the trigger service.
+Service accounts do not expire.
+
+However, both SA credentials and SA tokens have validity windows: 30 days for an SA credential and 1 day for an SA token.
```

Comment:
My understanding is that outside of the test mode, `ref-ledger-authenticator` talks to an underlying IAM, e.g., Auth0. Currently the docs seem to ignore that distinction but I think it’s important to point out which part is responsible for what and who defines things like the expiration dates (presumably the IAM and not ref-ledger-authenticator?).

Reply:
I believe the only difference with the test mode is that requests to `authorize` use basic auth (ignoring the password) instead of OAuth2, for which you would use a token issuer like Auth0. The validity windows are defined in a config file in the auth service itself.

Reply:
Isn’t auth0 signing the token? If so, I’m not sure how ref-ledger-authenticator can decide on the expiration date.

Reply:
I see, I think you're right. In the ref-ledger-authenticator there is a JwtJwkAuthenticator interface with a sample implementation that reads expiration details from a config file. I think for production it would be swapped out with Auth0, and I guess then those details would be configured separately.